src/DGtal/images/RigidTransformation2D.h

```diff
@@ -64,16 +65,17 @@ namespace functors
      *
      * @see exampleRigidtransformation2d.cpp
      */
-template <typename TSpace>
-class ForwardRigidTransformation2D
+template <typename TSpace, typename TFunctor, typename TInputValue, typename TOutputValue >
+class ForwardRigidTransformation2D : std::unary_function <TInputValue, TOutputValue>
```

Comment:
I think there may be two issues here:
1. `std::unary_function` is deprecated since `C++11` and is removed in `C++17` (see https://en.cppreference.com/w/cpp/utility/functional/unary_function). If inheriting from this class is necessary (for example in order to use some functions of `<functional>` header), I think we should better add the corresponding `argument_type` and `result_type` typedefs in this class. However, functions that need those typedefs tend to become deprecated in future C++ versions.
2. this modification add 3 template parameters that may broke current code (see the errors in Travis when compiling DGtalTools). I propose to add only the `TFunctor` template parameter, defaulted to an appropriate rounding functor (like `VectorRounding` you also add, or a `VectorFloor` if you want to keep current behavior). By making `operator()` templated, it should be possible to remove the need of `TInputValue` and `TOutputValue`.

The same remark for `ForwardRigitTransformation3D`.

Reply:
I think I have fixed this already.